### PR TITLE
use lnd rest api

### PR DESCRIPTION
**Pull request by antonilol**
#2214 but with rest api now, (lnd-api.ts looks way cleaner now lol)

### CODE REVIEW

**Review by softsimon — APPROVED** [review]
tested ACK @ v2.5.0-dev [[887fb13f](https://github.com/mempool/mempool/commit/887fb13f)]

File: backend/package.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
     "mempool",
     "blockchain",
     "explorer",
-    "liquid"
+    "liquid",
+    "lightning"
```
Comment (nymkappa) [inline backend/package.json:20]: We don't need this `lightning` dep anymore right?

File: backend/src/tasks/lightning/node-sync.service.ts [hunk 2]
```diff
@@ -418,8 +419,18 @@ class NodeSyncService {
...
+  private toShortId(id: string): string {
+    const n = BigInt(id);
+    return [
+      n >> 40n, // nth block
+      (n >> 16n) & 0xffffffn, // nth tx of the block
+      n & 0xffffn // nth output of the tx
+    ].join('x');
+  }
+
+  private utcDateToMysql(date?: number): string {
```
Comment (nymkappa) [inline backend/src/tasks/lightning/node-sync.service.ts:432]: I still don't understand what's that function for. `lnd` returns timestamp that we can directly manipulate and insert into the db without ever the need to convert it to string. @softsimon 

We can use `FROM_UNIXTIME(timestamp)` to insert in db.

**Review by nymkappa — CHANGES_REQUESTED** [review]

File: backend/package.json (hunk 1 above)
Comment (antonilol) [reply backend/package.json:20]: This one is an npm keyword/search word, the deps are a few more lines down

Comment (nymkappa) [reply backend/package.json:20]: Oh my bad!

File: backend/src/tasks/lightning/node-sync.service.ts (hunk 2 above)
Comment (nymkappa) [reply backend/src/tasks/lightning/node-sync.service.ts:432]: Actually nevermind I'll clean this up when I make the clightning PR

**Comment by nymkappa** [comment]
I got an issue running the PR, I'm not sure if it's related to the change or not, maybe the previous library was handling this because I never ran into it before?

```
Aug  1 10:28:48 [86969] ERR: <lightning> $scanForClosedChannels() error: connect ETIMEDOUT 127.0.0.1:8332
```

And now I have 2 channels close, and all others channels status is set to 0.

Screen Shot 2022-08-01 at 10 34 17 AM

**Review by nymkappa — APPROVED** [review]

**Review by wiz — APPROVED** [review]
Tested ACK @ v2.5.0-dev [[2a3111af](https://github.com/mempool/mempool/commit/2a3111af)]

Thanks for fixing this and removing the extra npm deps :+1:

**Pull request merged**